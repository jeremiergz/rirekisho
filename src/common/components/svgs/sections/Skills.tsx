import SVG, { SVGProps } from '@/common/components/misc/SVG';
import React from 'react';

function Skills(props: SkillsProps): JSX.Element {
  return (
    <SVG {...props}>
      <path d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z" />
    </SVG>
  );
}

export type SkillsProps = SVGProps;
export default Skills;
