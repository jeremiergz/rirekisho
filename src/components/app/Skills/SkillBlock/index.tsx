import React, { FunctionComponent } from 'react';
import Label from '../../../common/Label';
import Box, { BoxProps } from '../../../primitives/Box';
import FlexBox from '../../../primitives/FlexBox';
import Text from '../../../primitives/Text';
import Bubble from './Bubble';

const gradeBasis = 10;
const maxBubblesNumber = 7;

const SkillBlock: FunctionComponent<SkillBlockProps> = ({ items, title, ...rest }) => {
  const factor = gradeBasis / maxBubblesNumber;
  return (
    <Box marginBottom={4} maxWidth={{ _: 393, tablet: 356 }} width="100%" {...rest}>
      <Label title={title} />
      {items
        .sort((a, b) => a.sortOrder - b.sortOrder)
        .map(item => {
          const grade = Math.ceil(item.proficiencyLevel * factor);
          const fullBubblesNumber = [...Array(item.proficiencyLevel).keys()];
          const emptyBubblesNumber = [...Array(maxBubblesNumber - item.proficiencyLevel).keys()];
          return (
            <FlexBox alignItems="center" justifyContent="space-between" key={item.name} marginBottom={3}>
              <Text fontSize={18} fontWeight="bold">
                {item.name}
              </Text>
              <Text
                color="secondary"
                display={{ _: 'block', mobileL: 'none' }}
                fontSize={22}
                fontWeight="bold"
                lineHeight="16px"
              >
                {grade}/{gradeBasis}
              </Text>
              <FlexBox alignItems="center" display={{ _: 'none', mobileL: 'flex' }}>
                {fullBubblesNumber.map((bubble, index) => (
                  <Bubble key={bubble} marginRight={index !== maxBubblesNumber - 1 && 2} />
                ))}
                {emptyBubblesNumber.map((bubble, index) => (
                  <Bubble empty key={bubble} marginRight={index !== emptyBubblesNumber.length - 1 && 2} />
                ))}
              </FlexBox>
            </FlexBox>
          );
        })}
    </Box>
  );
};

SkillBlock.displayName = 'SkillBlock';

export type SkillBlockProps = BoxProps & {
  items: Models.Skill[];
  title: string;
};
export default SkillBlock;
