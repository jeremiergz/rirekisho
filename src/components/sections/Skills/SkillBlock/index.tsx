import Label from '@/components/common/Label';
import clsx from 'clsx';
import React from 'react';
import Bubble from './Bubble';

const gradeBasis = 10;
const maxBubblesNumber = 7;

function SkillBlock({ className, items, style, title, ...rest }: SkillBlockProps): JSX.Element {
  const factor = gradeBasis / maxBubblesNumber;

  return (
    <div className={clsx(className, 'mb-6 w-full')} {...rest} style={{ ...style, maxWidth: 373 }}>
      <Label title={title} />
      {items.map((item, index) => {
        const grade = Math.ceil(item.proficiencyLevel * factor);
        const fullBubblesNumber = [...Array(item.proficiencyLevel).keys()];
        const emptyBubblesNumber = [...Array(maxBubblesNumber - item.proficiencyLevel).keys()];
        const isLast = index === items.length - 1;

        return (
          <div className={clsx('flex items-center justify-between', isLast || 'mb-2')} key={item.name}>
            <span className="font-bold text-lg text-gray-900 dark:text-gray-300 mr-6">{item.name}</span>
            <span className="font-bold sm:hidden leading-4 text-secondary text-xl">
              {grade}/{gradeBasis}
            </span>
            <div className="hidden sm:flex items-center">
              {fullBubblesNumber.map((bubble, index) => (
                <Bubble className={clsx(index !== maxBubblesNumber - 1 && 'mr-2')} key={bubble} />
              ))}
              {emptyBubblesNumber.map((bubble, index) => (
                <Bubble className={clsx(index !== emptyBubblesNumber.length - 1 && 'mr-2')} empty key={bubble} />
              ))}
            </div>
          </div>
        );
      })}
    </div>
  );
}

export type SkillBlockProps = React.ComponentProps<'div'> & {
  items: Models.SkillItem[];
  title: string;
};
export default SkillBlock;
