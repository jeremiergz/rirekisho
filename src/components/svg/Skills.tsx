import React, { ComponentProps } from 'react';

const Skills: React.FC<SkillsProps> = props => {
  return (
    <svg fill="#fff" height="24" width="24" {...props}>
      <path d="M0 0h24v24H0V0z" fill="none"></path>
      <path d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"></path>
    </svg>
  );
};

Skills.displayName = 'Skills';

export type SkillsProps = ComponentProps<'svg'>;
export default Skills;
