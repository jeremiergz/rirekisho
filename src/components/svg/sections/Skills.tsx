import React, { ComponentProps } from 'react';

const Skills: React.FC<SkillsProps> = ({ fill, ...rest }) => {
  return (
    <svg fill={fill} viewBox="0 0 24 24" {...rest}>
      <path d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z" />
    </svg>
  );
};

Skills.displayName = 'Skills';

export type SkillsProps = ComponentProps<'svg'>;
export default Skills;
