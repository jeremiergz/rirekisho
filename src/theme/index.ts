const breakpoints = ['320px', '375px', '425px', '768px', '1024px', '1200px', '1440px', '2560px'] as const;
Object.defineProperties(breakpoints, {
  mobileS: { value: breakpoints[0] },
  mobileM: { value: breakpoints[1] },
  mobileL: { value: breakpoints[2] },
  tablet: { value: breakpoints[3] },
  laptopS: { value: breakpoints[4] },
  laptopM: { value: breakpoints[5] },
  laptopL: { value: breakpoints[6] },
  desktop: { value: breakpoints[7] },
});

const colors = {
  background: 'white',
  dark: '#212121',
  error: '#f44336',
  info: '#2196f3',
  primary: '#17365c',
  secondary: '#31859a',
  success: '#4caf50',
  text: '#262626',
  white: '#ffffff',
} as const;

const fonts = {
  main: "'Open Sans', sans-serif",
  title: "'Baloo Bhaina 2', cursive",
} as const;

const fontWeights = {
  bold: 800,
  bolder: 600,
  regular: 400,
} as const;

const theme = {
  breakpoints,
  colors,
  fonts,
  fontWeights,
} as const;

type Theme = typeof theme;

export type { Theme };
export default theme;
