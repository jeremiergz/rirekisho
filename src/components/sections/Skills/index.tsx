import Layout from 'components/Layout';
import { graphql, useStaticQuery } from 'gatsby';
import React, { forwardRef } from 'react';
import SkillBlock from './SkillBlock';

const Skills = forwardRef<HTMLDivElement>((_, ref) => {
  const {
    skillNodes: { nodes: skills },
  } = useStaticQuery<GraphQL.SkillsData>(graphql`
    query SkillsData {
      skillNodes: allSkillsJson {
        nodes {
          items {
            name
            proficiencyLevel
            section
            sortOrder
          }
          name
          sortOrder
        }
      }
    }
  `);
  return (
    <Layout.Section ref={ref} title="skills">
      <Layout.Content
        justifyContent={{ _: 'space-around', laptopL: 'space-between' }}
        marginX={{ _: 0, tablet: -3 }}
        width={{ _: 'auto', tablet: 'calc(100% + 32px)' }}
      >
        {skills
          .sort((a, b) => a.sortOrder - b.sortOrder)
          .map(item => (
            <SkillBlock key={item.name} items={item.items} marginX={3} title={item.name} />
          ))}
      </Layout.Content>
    </Layout.Section>
  );
});

Skills.displayName = 'Skills';

export default Skills;
