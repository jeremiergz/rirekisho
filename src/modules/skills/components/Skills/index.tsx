import Content from '@/common/components/layout/Content';
import Section from '@/common/components/layout/Section';
import clsx from 'clsx';
import { forwardRef } from 'react';
import useSkillsData from '../../hooks/useSkillsData';
import SkillBlock from './SkillBlock';

const Skills = forwardRef<HTMLDivElement>(function Skills(_, ref): JSX.Element {
  const skills = useSkillsData();

  // FIXME: Display only supports 3 skills groups
  const [skillsGroup1, skillsGroup2, skillsGroup3] = skills.sort((a, b) => a.sortOrder - b.sortOrder);

  return (
    <Section className="print:mt-2" ref={ref} title="skills">
      <Content className="justify-center">
        <SkillBlock
          className={clsx('mx-4 lg:ml-0', 'print:mx-0')}
          key={skillsGroup1.name}
          items={skillsGroup1.items}
          title={skillsGroup1.name}
        />
        <SkillBlock
          className={clsx('mx-4', 'print:mx-4')}
          key={skillsGroup2.name}
          items={skillsGroup2.items}
          title={skillsGroup2.name}
        />
        <SkillBlock
          className={clsx('mx-4 lg:mr-0', 'print:mx-0')}
          key={skillsGroup3.name}
          items={skillsGroup3.items}
          title={skillsGroup3.name}
        />
      </Content>
    </Section>
  );
});

export default Skills;
