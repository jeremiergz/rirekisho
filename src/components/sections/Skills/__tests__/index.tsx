import * as Gatsby from 'gatsby';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from 'theme';
import Skills from '..';
import mockData from './mock.json';

describe('components/sections/Skills Test Suite', () => {
  beforeAll(() => {
    jest.spyOn(Gatsby, 'graphql').mockImplementation(() => '');
    jest.spyOn(Gatsby, 'useStaticQuery').mockImplementation(() => mockData);
  });

  it('matches snapshot', () => {
    const jsx = (
      <ThemeProvider theme={theme}>
        <Skills />
      </ThemeProvider>
    );
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
