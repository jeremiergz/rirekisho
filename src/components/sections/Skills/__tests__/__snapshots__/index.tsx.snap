// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/sections/Skills Test Suite matches snapshot 1`] = `
<section
  className="sc-AxhCb ipofvh"
>
  <div
    className="sc-AxirZ bLeYFs"
    fontSize={28}
    fontWeight="bold"
  >
    <span
      className="sc-AxiKw ftGYeL"
      color="secondary"
    >
      ski
    </span>
    <span
      className="sc-AxiKw emkGCh"
      color="primary"
    >
      lls
    </span>
  </div>
  <div
    className="sc-AxjAm eSgDAg"
    width={
      Object {
        "_": "auto",
        "tablet": "calc(100% + 32px)",
      }
    }
  >
    <div
      className="sc-AxirZ clLqNQ"
      width="100%"
    >
      <label
        className="sc-AxiKw eioTIl"
        color="secondary"
        fontSize={14}
        fontWeight="bold"
      >
        programming
      </label>
      <div
        className="sc-AxjAm ikStGQ"
      >
        <span
          className="sc-AxiKw eUCEIz"
          fontSize={18}
          fontWeight="bold"
        >
          &lt;Java|Type&gt;Script
        </span>
        <span
          className="sc-AxiKw UyLlj"
          color="secondary"
          display={
            Object {
              "_": "block",
              "mobileL": "none",
            }
          }
          fontSize={22}
          fontWeight="bold"
        >
          10
          /
          10
        </span>
        <div
          className="sc-AxjAm hcQnsb"
          display={
            Object {
              "_": "none",
              "mobileL": "flex",
            }
          }
        >
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ kFNaJk"
            height={16}
            width={16}
          />
        </div>
      </div>
      <div
        className="sc-AxjAm ikStGQ"
      >
        <span
          className="sc-AxiKw eUCEIz"
          fontSize={18}
          fontWeight="bold"
        >
          Node.js
        </span>
        <span
          className="sc-AxiKw UyLlj"
          color="secondary"
          display={
            Object {
              "_": "block",
              "mobileL": "none",
            }
          }
          fontSize={22}
          fontWeight="bold"
        >
          10
          /
          10
        </span>
        <div
          className="sc-AxjAm hcQnsb"
          display={
            Object {
              "_": "none",
              "mobileL": "flex",
            }
          }
        >
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ kFNaJk"
            height={16}
            width={16}
          />
        </div>
      </div>
      <div
        className="sc-AxjAm ikStGQ"
      >
        <span
          className="sc-AxiKw eUCEIz"
          fontSize={18}
          fontWeight="bold"
        >
          Golang
        </span>
        <span
          className="sc-AxiKw UyLlj"
          color="secondary"
          display={
            Object {
              "_": "block",
              "mobileL": "none",
            }
          }
          fontSize={22}
          fontWeight="bold"
        >
          6
          /
          10
        </span>
        <div
          className="sc-AxjAm hcQnsb"
          display={
            Object {
              "_": "none",
              "mobileL": "flex",
            }
          }
        >
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ cihBLs"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ cihBLs"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ ksRrQS"
            height={16}
            width={16}
          />
        </div>
      </div>
      <div
        className="sc-AxjAm ikStGQ"
      >
        <span
          className="sc-AxiKw eUCEIz"
          fontSize={18}
          fontWeight="bold"
        >
          HTML5|CSS3
        </span>
        <span
          className="sc-AxiKw UyLlj"
          color="secondary"
          display={
            Object {
              "_": "block",
              "mobileL": "none",
            }
          }
          fontSize={22}
          fontWeight="bold"
        >
          9
          /
          10
        </span>
        <div
          className="sc-AxjAm hcQnsb"
          display={
            Object {
              "_": "none",
              "mobileL": "flex",
            }
          }
        >
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ ksRrQS"
            height={16}
            width={16}
          />
        </div>
      </div>
    </div>
    <div
      className="sc-AxirZ clLqNQ"
      width="100%"
    >
      <label
        className="sc-AxiKw eioTIl"
        color="secondary"
        fontSize={14}
        fontWeight="bold"
      >
        devops
      </label>
      <div
        className="sc-AxjAm ikStGQ"
      >
        <span
          className="sc-AxiKw eUCEIz"
          fontSize={18}
          fontWeight="bold"
        >
          Kubernetes
        </span>
        <span
          className="sc-AxiKw UyLlj"
          color="secondary"
          display={
            Object {
              "_": "block",
              "mobileL": "none",
            }
          }
          fontSize={22}
          fontWeight="bold"
        >
          8
          /
          10
        </span>
        <div
          className="sc-AxjAm hcQnsb"
          display={
            Object {
              "_": "none",
              "mobileL": "flex",
            }
          }
        >
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ cihBLs"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ ksRrQS"
            height={16}
            width={16}
          />
        </div>
      </div>
      <div
        className="sc-AxjAm ikStGQ"
      >
        <span
          className="sc-AxiKw eUCEIz"
          fontSize={18}
          fontWeight="bold"
        >
          Docker
        </span>
        <span
          className="sc-AxiKw UyLlj"
          color="secondary"
          display={
            Object {
              "_": "block",
              "mobileL": "none",
            }
          }
          fontSize={22}
          fontWeight="bold"
        >
          10
          /
          10
        </span>
        <div
          className="sc-AxjAm hcQnsb"
          display={
            Object {
              "_": "none",
              "mobileL": "flex",
            }
          }
        >
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ kFNaJk"
            height={16}
            width={16}
          />
        </div>
      </div>
      <div
        className="sc-AxjAm ikStGQ"
      >
        <span
          className="sc-AxiKw eUCEIz"
          fontSize={18}
          fontWeight="bold"
        >
          &lt;GitLab|Travis&gt;CI
        </span>
        <span
          className="sc-AxiKw UyLlj"
          color="secondary"
          display={
            Object {
              "_": "block",
              "mobileL": "none",
            }
          }
          fontSize={22}
          fontWeight="bold"
        >
          10
          /
          10
        </span>
        <div
          className="sc-AxjAm hcQnsb"
          display={
            Object {
              "_": "none",
              "mobileL": "flex",
            }
          }
        >
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ kFNaJk"
            height={16}
            width={16}
          />
        </div>
      </div>
      <div
        className="sc-AxjAm ikStGQ"
      >
        <span
          className="sc-AxiKw eUCEIz"
          fontSize={18}
          fontWeight="bold"
        >
          Ansible
        </span>
        <span
          className="sc-AxiKw UyLlj"
          color="secondary"
          display={
            Object {
              "_": "block",
              "mobileL": "none",
            }
          }
          fontSize={22}
          fontWeight="bold"
        >
          9
          /
          10
        </span>
        <div
          className="sc-AxjAm hcQnsb"
          display={
            Object {
              "_": "none",
              "mobileL": "flex",
            }
          }
        >
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ ksRrQS"
            height={16}
            width={16}
          />
        </div>
      </div>
    </div>
    <div
      className="sc-AxirZ clLqNQ"
      width="100%"
    >
      <label
        className="sc-AxiKw eioTIl"
        color="secondary"
        fontSize={14}
        fontWeight="bold"
      >
        frameworks
      </label>
      <div
        className="sc-AxjAm ikStGQ"
      >
        <span
          className="sc-AxiKw eUCEIz"
          fontSize={18}
          fontWeight="bold"
        >
          &lt;Koa|Express&gt;.js
        </span>
        <span
          className="sc-AxiKw UyLlj"
          color="secondary"
          display={
            Object {
              "_": "block",
              "mobileL": "none",
            }
          }
          fontSize={22}
          fontWeight="bold"
        >
          10
          /
          10
        </span>
        <div
          className="sc-AxjAm hcQnsb"
          display={
            Object {
              "_": "none",
              "mobileL": "flex",
            }
          }
        >
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ kFNaJk"
            height={16}
            width={16}
          />
        </div>
      </div>
      <div
        className="sc-AxjAm ikStGQ"
      >
        <span
          className="sc-AxiKw eUCEIz"
          fontSize={18}
          fontWeight="bold"
        >
          React.js
        </span>
        <span
          className="sc-AxiKw UyLlj"
          color="secondary"
          display={
            Object {
              "_": "block",
              "mobileL": "none",
            }
          }
          fontSize={22}
          fontWeight="bold"
        >
          9
          /
          10
        </span>
        <div
          className="sc-AxjAm hcQnsb"
          display={
            Object {
              "_": "none",
              "mobileL": "flex",
            }
          }
        >
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ ksRrQS"
            height={16}
            width={16}
          />
        </div>
      </div>
      <div
        className="sc-AxjAm ikStGQ"
      >
        <span
          className="sc-AxiKw eUCEIz"
          fontSize={18}
          fontWeight="bold"
        >
          Jest|Mocha|Chai
        </span>
        <span
          className="sc-AxiKw UyLlj"
          color="secondary"
          display={
            Object {
              "_": "block",
              "mobileL": "none",
            }
          }
          fontSize={22}
          fontWeight="bold"
        >
          9
          /
          10
        </span>
        <div
          className="sc-AxjAm hcQnsb"
          display={
            Object {
              "_": "none",
              "mobileL": "flex",
            }
          }
        >
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ ksRrQS"
            height={16}
            width={16}
          />
        </div>
      </div>
      <div
        className="sc-AxjAm ikStGQ"
      >
        <span
          className="sc-AxiKw eUCEIz"
          fontSize={18}
          fontWeight="bold"
        >
          Gatsby.js|Next.js
        </span>
        <span
          className="sc-AxiKw UyLlj"
          color="secondary"
          display={
            Object {
              "_": "block",
              "mobileL": "none",
            }
          }
          fontSize={22}
          fontWeight="bold"
        >
          9
          /
          10
        </span>
        <div
          className="sc-AxjAm hcQnsb"
          display={
            Object {
              "_": "none",
              "mobileL": "flex",
            }
          }
        >
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ jPxvnW"
            height={16}
            width={16}
          />
          <span
            className="sc-AxirZ ksRrQS"
            height={16}
            width={16}
          />
        </div>
      </div>
    </div>
  </div>
</section>
`;
